internal/service/iam/service_linked_role_test.go

```diff
@@ -88,18 +88,13 @@ func TestAccIAMServiceLinkedRole_basic(t *testing.T) {
 			{
 				PreConfig: func() {
 					// Remove existing if possible
-					conn := acctest.Provider.Meta().(*conns.AWSClient).IAMConn
-					deletionID, err := tfiam.DeleteServiceLinkedRole(conn, name)
-					if err != nil {
-						t.Fatalf("Error deleting service-linked role %s: %s", name, err)
-					}
-					if deletionID == "" {
-						return
-					}
+					r := tfiam.ResourceServiceLinkedRole()
```

Comment:
Nice! I was wreaking my head on how to do
this now I see i didn’t pass the right context..

Reply:
Actually, need to create an ARN - fixed that now.